options.go

```diff
@@ -234,6 +235,24 @@ func WithIAMAuthN() Option {
 	}
 }
 
+// NetResolver groups the methods on net.Resolver that are used by the DNS
+// resolver implementation. This allows an application to replace the default
+// net.DefaultResolver with a custom implementation. For example: the
+// application may need to connect to a specific DNS server using a specially
+// configured instance of net.Resolver.
+type NetResolver interface {
+	LookupSRV(ctx context.Context, service, proto, name string) (cname string, addrs []*net.SRV, err error)
```

Comment:
The more I think about this, the more I think this is the wrong interface. What we want generically is a way to go from some name (DNS record or key value store path or whatever the user wants) to an instance connection name. This would be more future proof for when we want to support alternate resolution techniques

``` golang
type InstanceConnectionNameResolver interface {
    Resolve(ctx context.Context, name string) (string, error) 
}
```

Then we could provide a default DNS resolve that does the equivalent of the SRV lookup. I think this is a small change, but will unlock many more use cases (especially as we discover the shortcomings of DNS polling).

Reply:
Good insight, I agree. The Java implementation started heading in this direction too. I will update the PR accordingly.